packages/babel-parser/src/parser/expression.js

```diff
@@ -380,6 +380,58 @@ export default class ExpressionParser extends LValParser {
 
         node.right = this.parseExprOpRightExpr(op, prec, noIn);
 
+        /* this check is needed because for operations like
+         * a ?? b || c
+         * a ?? b => This is considered as a logical expression in the ast tree
+         * c => Identifier
+         * so we need to check, only for || cases that if left is logical expression without paren
+         * then raise an error
+         */
+
+        if (op === tt.logicalOR) {
+          if (
+            left.type === "LogicalExpression" &&
+            left.operator === "??" &&
+            !(left.extra && left.extra.parenthesized)
+          ) {
+            throw this.raise(
+              left.start,
+              `Wrap left hand side of the expression in parenthesis`,
+            );
+          }
+        }
+        /* this check is for all && and ?? operators
+         * a ?? b && c for this example
+         * b && c => This is considered as a logical expression in the ast tree
+         * a => Identifier
+         * so for ?? operator we need to check in this case the right expression
+         * second case a && b ?? c
+         * here a && b => This is considered as a logical expression in the ast tree
+         * c => identifer
+         * so now here for ?? operator we need to check the left expression
+         */
+        if (op === tt.nullishCoalescing) {
+          if (
+            left.type === "LogicalExpression" &&
+            left.operator !== "??" &&
+            !(left.extra && left.extra.parenthesized)
+          ) {
+            throw this.raise(
+              left.start,
+              `Wrap left hand side of the expression in parenthesis`,
```

Comment:
Let's change this message, since it's not clear if the `??` was meant to be a group or the other operator was meant to be a group. I'd suggest "Nullish coalescing operator requires parens when mixing with logical operators", but I'm not sure end users will understand what nullish coalescing operator is or what the other logical operators are.

For reference, the exponentiation operator uses "Wrap left hand side or entire exponentiation in parentheses."

Reply:
Thank you @jridgewell. As per your suggestion adding this as a message
```
Nullish coalescing operator(??) requires parens when mixing with logical operators
```